### DISCUSSION TITLE
Use math.fractions in math.vec

### DISCUSSION THREAD

**Discussion by jorgeluismireles** [vlang/v#21994] [Questions and Answers]
I found very interesting the module `math.fractions` to be used in [rational number](https://en.wikipedia.org/wiki/Rational_number) problems.

First I tried to use the fractions in vectors of module `math.vec` and got a lot of errors. Then I reduce the scope and notice `math.max()` works but `math.abs()` gives two errors. The code:
```
import math
import math.fractions

fn main() {
        a := fractions.fraction(2, 8)
        b := fractions.fraction(2, 10)

        println('${a} + ${b} : ${a+b}') // ok
        println('${a} >= ${b} : ${ a >= b }') // ok
        println('reduce(${a}) : ${a.reduce()}') // ok
        println('max(${a},${b}) : ${math.max(a,b)}') // ok

        println('abs(${a}) : ${math.abs(a)}') // errors
}
```
The errors:
```
/home/jorge/v/vlib/math/min_max_abs.v:21:12: error: infix expr: cannot use `int literal` (right expression) as `math.fractions.Fraction`
   19 | @[inline]
   20 | pub fn abs[T](a T) T {
   21 |     return if a < 0 { -a } else { a }
      |               ~~~~~
   22 | }
/home/jorge/v/vlib/math/min_max_abs.v:21:20: error: operator `-` can only be used with numeric types, but the value after `-` is of type `math.fractions.Fraction` instead
   19 | @[inline]
   20 | pub fn abs[T](a T) T {
   21 |     return if a < 0 { -a } else { a }
      |                       ^
   22 | }
```
I wonder how hard is for the `fractions` to "implement" the zero comparison and the negative operator. And for the case to be used as members in vectors, I notice a lot of errors regarding cannot be square rooted among other operations used inside `math.vec`.

Regards

**Comment by JalonSolov** [comment]
Use `a < 0.0` and `0.0 - a`

This looks more like a problem related to type promotion... `0` should always equal `0`... `0.0`, `0e0`, `1 - 1`, etc., etc.
